src/compiler/tsbuildPublic.ts

```diff
@@ -1333,9 +1356,104 @@ namespace ts {
         return { buildResult, step: BuildStep.QueueReferencingProjects };
     }
 
+    function isFileWatcherWithModifiedTime(value: FileWatcherWithModifiedTime | Date): value is FileWatcherWithModifiedTime {
+        return !!(value as FileWatcherWithModifiedTime).watcher;
+    }
+
+    function getModifiedTime(state: SolutionBuilderState, fileName: string): Date {
+        const path = toPath(state, fileName);
+        const existing = state.filesWatched.get(path);
+        if (state.watch && !!existing) {
+            if (!isFileWatcherWithModifiedTime(existing)) return existing;
+            if (existing.modifiedTime) return existing.modifiedTime;
+        }
+        const result = ts.getModifiedTime(state.host, fileName);

```

Comment:
This is where we store modified time for input files.. 
This is either `Date | FileWatcherWithModifiedTime` because we query modified times first and then after complete compilation of the project, watch the files so we dont want to loose these modified times.